stubs/pysftp/pysftp/__init__.pyi

```diff
@@ -0,0 +1,125 @@
+    ) -> None: ...
+    def get_d(self, remotedir: _Path, localdir: _Path, preserve_mtime: bool = ...) -> None: ...
+    def get_r(self, remotedir: _Path, localdir: _Path, preserve_mtime: bool = ...) -> None: ...
+    def getfo(self, remotepath: _Path, flo: IO[bytes], callback: Optional[_Callback] = ...) -> int: ...
+    def put(
+        self,
+        localpath: _Path,
+        remotepath: Optional[_Path] = ...,
+        callback: Optional[_Callback] = ...,
+        confirm: bool = ...,
+        preserve_mtime: bool = ...,
+    ) -> paramiko.SFTPAttributes: ...
+    def put_d(self, localpath: _Path, remotepath: _Path, confirm: bool = ..., preserve_mtime: bool = ...) -> None: ...
+    def put_r(self, localpath: _Path, remotepath: _Path, confirm: bool = ..., preserve_mtime: bool = ...) -> None: ...
+    def putfo(
+        self,
+        flo: IO[bytes],
+        remotepath: Optional[_Path] = ...,
+        file_size: int = ...,
+        callback: Optional[_Callback] = ...,
+        confirm: bool = ...,
+    ) -> paramiko.SFTPAttributes: ...
+    def execute(self, command: str) -> List[str]: ...
+    def cd(self, remotepath: Optional[_Path] = ...) -> None: ...  # noqa: F811
+    def chdir(self, remotepath: _Path) -> None: ...
+    def cwd(self, remotepath: _Path) -> None: ...
+    def chmod(self, remotepath: _Path, mode: int = ...) -> None: ...
+    def chown(self, remotepath: _Path, uid: Optional[int] = ..., gid: Optional[int] = ...) -> None: ...
+    def getcwd(self) -> str: ...
+    def listdir(self, remotepath: _Path = ...) -> List[str]: ...
+    def listdir_attr(self, remotepath: _Path = ...) -> List[paramiko.SFTPAttributes]: ...
+    def mkdir(self, remotepath: _Path, mode: int = ...) -> None: ...
+    def normalize(self, remotepath: _Path) -> str: ...
+    def isdir(self, remotepath: _Path) -> bool: ...
+    def isfile(self, remotepath: _Path) -> bool: ...
+    def makedirs(self, remotedir: _Path, mode: int = ...) -> None: ...
+    def readlink(self, remotelink: _Path) -> str: ...
+    def remove(self, remotefile: _Path) -> None: ...
+    def unlink(self, remotefile: _Path) -> None: ...
+    def rmdir(self, remotepath: _Path) -> None: ...
+    def rename(self, remote_src: _Path, remote_dest: _Path) -> None: ...
+    def stat(self, remotepath: _Path) -> paramiko.SFTPAttributes: ...
+    def lstat(self, remotepath: _Path) -> paramiko.SFTPAttributes: ...
+    def close(self) -> None: ...
+    def open(self, remote_file: _Path, mode: str = ..., bufsize: int = ...) -> paramiko.SFTPFile: ...
+    def exists(self, remotepath: _Path) -> bool: ...
+    def lexists(self, remotepath: _Path) -> bool: ...
+    def symlink(self, remote_src: _Path, remote_dest: _Path) -> None: ...
+    def truncate(self, remotepath: _Path, size: int) -> int: ...
+    def walktree(  # noqa: F811
+        self, remotepath: _Path, fcallback: _Callback, dcallback: _Callback, ucallback: _Callback, recurse: bool = ...
```

Comment:
These are not _Callback, they're called with a single str argument for the path. (Or a bytes arguments if remotepath was bytes.) This should presumably match TreeItemCallback in helpers.pyi; you can import it from there.